File: frontend/src/app/models/game-data-dto.ts
import { BuffDefinitionDTO } from "./buff-definition-dto";
import { ClassStatsDTO } from "./class-stats-dto";
import { DragonStoneDefinitionDTO } from "./dragon-stone-definition-dto";
import { EnchantmentDTO } from "./enchantment-dto";
import { EssenceDefinitionDTO } from "./essence-definition-dto";
import { GemDefinitionDTO } from "./gem-definition-dto";
import { ItemDefinitionDTO } from "./item-definition-dto";
import { JewelDefinitionDTO } from "./jewel-definition-dto";
import { LevelMultiplierTableDTO } from "./level-multiplier-table-dto";
import { PetDefinitionDTO } from "./pet-definition-dto";
import { RuneDefinitionDTO } from "./rune-definition-dto";
import { SetDTO } from "./set-dto";
import { WisdomSkillTreeDefinitionDTO } from "./wisdom-skill-tree-dto";

export interface GameDataDTO {
    characterClassStats:  Record<string, ClassStatsDTO>;
    items: Record<string, ItemDefinitionDTO[]>;
    sets: Record<string, SetDTO[]>;
    jewels: Record<string, JewelDefinitionDTO[]>;
    enchantments: EnchantmentDTO[];
    gems: GemDefinitionDTO[];
    runes: RuneDefinitionDTO[];
    dragonStones: DragonStoneDefinitionDTO[];
    pets: PetDefinitionDTO[];
    essences: EssenceDefinitionDTO[];
    tonics: BuffDefinitionDTO[];
    physics: BuffDefinitionDTO[];
    levelMultiplierTable: LevelMultiplierTableDTO;
    wisdomSkillTree: WisdomSkillTreeDefinitionDTO;
}

File: src/main/java/com/langleon/dsobuildsim/gamedata/dto/GameDataDTO.java
package com.langleon.dsobuildsim.gamedata.dto;

import com.langleon.dsobuildsim.buffs.dto.BuffDefinitionDTO;
import com.langleon.dsobuildsim.character.CharacterClass;
import com.langleon.dsobuildsim.dragonstones.dto.DragonStoneDefinitionDTO;
import com.langleon.dsobuildsim.items.dto.ItemDefinitionDTO;
import com.langleon.dsobuildsim.wisdomskilltree.dto.definition.WisdomSkillTreeDefinitionDTO;
import com.langleon.dsobuildsim.enchantments.dto.EnchantmentDTO;
import com.langleon.dsobuildsim.essences.dto.EssenceDefinitionDTO;
import com.langleon.dsobuildsim.gems.dto.GemDefinitionDTO;
import com.langleon.dsobuildsim.jewels.dto.JewelDefinitionDTO;
import com.langleon.dsobuildsim.pets.dto.PetDefinitionDTO;
import com.langleon.dsobuildsim.runes.dto.RuneDefinitionDTO;
import com.langleon.dsobuildsim.sets.dto.SetDTO;

import java.util.List;
import java.util.Map;

public record GameDataDTO (
        Map<CharacterClass, ClassStatsDTO> characterClassStats,
        Map<CharacterClass, List<ItemDefinitionDTO>> items,
        Map<CharacterClass, List<SetDTO>> sets,
        Map<CharacterClass, List<JewelDefinitionDTO>> jewels,
        List<EnchantmentDTO> enchantments,
        List<GemDefinitionDTO> gems,
        List<RuneDefinitionDTO> runes,
        List<DragonStoneDefinitionDTO> dragonStones,
        List<PetDefinitionDTO> pets,
        List<EssenceDefinitionDTO> essences,
        List<BuffDefinitionDTO> tonics,
        List<BuffDefinitionDTO> physics,
        LevelMultiplierTableDTO levelMultiplierTable,
        WisdomSkillTreeDefinitionDTO wisdomSkillTree
        )
{
}
